### PR TITLE
autoconf: provide a harfbuzz-config for use from CMake

### PR DESCRIPTION
**Pull request by mathstuf**
---
Cc: @ebraminio

### CODE REVIEW

File: src/harfbuzz-config.cmake.in [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+set(_harfbuzz_version_info "@HB_LIBTOOL_VERSION_INFO@")
+string(REPLACE ":" ";" _harfbuzz_version_info "${_harfbuzz_version_info}")
+list(GET _harfbuzz_version_info 0
+  _harfbuzz_current)
+list(GET _harfbuzz_version_info 1
+  _harfbuzz_revision)
+list(GET _harfbuzz_version_info 2
+  _harfbuzz_age)
+unset(_harfbuzz_version_info)
+math(EXPR _harfbuzz_soname "${_harfbuzz_current} - ${_harfbuzz_age}")
```
Comment (mathstuf) [inline src/harfbuzz-config.cmake.in]: I don't know that this is true. I don't see something obvious which has the soname string.

Comment (mathstuf) [reply src/harfbuzz-config.cmake.in]: Digging into libtool code, this actually does seem to be true except for:

  - freebsd-aout
  - qnx
  - sco
  - sunos

**Comment by mathstuf** [comment]
Tested with a simple CMake project:

```cmake
cmake_minimum_required(VERSION 3.3)
project(foo C)

find_package(harfbuzz REQUIRED)

add_executable(foo foo.c)
target_link_libraries(foo
    harfbuzz::icu)
```

```c
int main(int argc, char* argv[]) { return 0; }
```

File: src/harfbuzz-config.cmake.in (hunk 1 above)
Comment (behdad) [reply src/harfbuzz-config.cmake.in]: Our soname will stay 0 forever. If that ever was to change, the library will be called something else, like harfbuzz2.

Comment (mathstuf) [reply src/harfbuzz-config.cmake.in]: Sounds good to me.

**Comment by ebraminio** [comment]
Thank you for this! :)

File: src/harfbuzz-config.cmake.in [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  _harfbuzz_revision)
+list(GET _harfbuzz_version_info 2
+  _harfbuzz_age)
+unset(_harfbuzz_version_info)
+
+# Add the libraries.
+add_library(harfbuzz::harfbuzz SHARED IMPORTED)
+set_target_properties(harfbuzz::harfbuzz PROPERTIES
+  INTERFACE_INCLUDE_DIRECTORIES "${_harfbuzz_prefix}/${_harfbuzz_includedir}/harfbuzz"
+  IMPORTED_LOCATION "${_harfbuzz_prefix}/${_harfbuzz_libdir}/libharfbuzz.so.0.${_harfbuzz_current}.${_harfbuzz_revision}")
```
Comment (ebraminio) [inline src/harfbuzz-config.cmake.in]: Is this working on macOS with `.dylib`s?

Comment (mathstuf) [reply src/harfbuzz-config.cmake.in]: Nope. I can try that out tomorrow.

Comment (ebraminio) [reply src/harfbuzz-config.cmake.in]: Thank you `.so` much :P

**Comment by mathstuf** [comment]
Now tested on macOS as well.

**Comment by ebraminio** [comment]
@mathstuf, thank you very much!

**Pull request merged**